gcore/gdaldrivermanager.cpp

```diff
@@ -1146,6 +1219,496 @@ void GDALDriverManager::ReorderDrivers()
 #endif
 }
 
+/************************************************************************/
+/*                       GDALPluginDriverProxy                          */
+/************************************************************************/
+
+/** Constructor for a plugin driver proxy.
+ *
+ * @param osPluginFileName Plugin filename. e.g "ogr_Parquet.so"
+ */
+GDALPluginDriverProxy::GDALPluginDriverProxy(
+    const std::string &osPluginFileName)
+    : m_osPluginFileName(osPluginFileName)
+{
+}
+
+//! @cond Doxygen_Suppress
+GDALDriver::OpenCallback GDALPluginDriverProxy::GetOpenCallback()
+{
+    auto poRealDriver = GetRealDriver();
```

Comment:
I'm not  big fan of macros but maybe here it could be used to reduce duplicated checks, maybe something like ENSURE_REAL_DRIVER

Reply:
Good suggestion. Code now looks like:
```
#define DEFINE_DRIVER_METHOD_GET_CALLBACK(method_name, output_type)            \
    GDALDriver::output_type GDALPluginDriverProxy::method_name()               \
    {                                                                          \
        auto poRealDriver = GetRealDriver();                                   \
        if (!poRealDriver)                                                     \
            return nullptr;                                                    \
        return poRealDriver->method_name();                                    \
    }

DEFINE_DRIVER_METHOD_GET_CALLBACK(GetOpenCallback, OpenCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetCreateCallback, CreateCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetCreateMultiDimensionalCallback,
                                  CreateMultiDimensionalCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetCreateCopyCallback, CreateCopyCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetDeleteCallback, DeleteCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetRenameCallback, RenameCallback)
DEFINE_DRIVER_METHOD_GET_CALLBACK(GetCopyFilesCallback, CopyFilesCallback)
```